<launch>
  <arg name="machine" default="c2"/>
  <arg name="use_mask" default="false"/>

  <arg name="input_image" doc="input camera image topic"/>
  <arg name="input_camera_info" doc="input camera info topic"/>

  <node name="fridge_demo_sift" pkg="imagesift" type="imagesift"
        machine="$(arg machine)">
    <!--         launch-prefix="nice -n +10" -->
    <remap from="image" to="$(arg input_image)" />
    <remap from="camera_info" to="$(arg input_camera_info)"/>
    <remap from="mask" to="attention_clipper/output/mask"/>
    <param name="use_mask" value="$(arg use_mask)"/>
  </node>
  <node name="fridge_detector" pkg="jsk_perception" type="point_pose_extractor"
        machine="$(arg machine)">
    <!-- NOTE: rosparam need to be overwritten by parent launch file -->
  </node>
  <node name="fridge_detector_agg" pkg="posedetection_msgs" type="object_detection_aggregator"
        machine="$(arg machine)">
    <rosparam>
      pub_rate: 2.0
    </rosparam>
  </node>
  
  <group if="$(arg use_mask)">
    <node pkg="jsk_pcl_ros" type="attention_clipper" name="attention_clipper"
          output="screen" machine="$(arg machine)">
      <remap from="~input" to="$(arg input_camera_info)" />
      <rosparam>
        dimension_x: 0.3
        dimension_y: 0.3
        dimension_z: 0.05
        frame_id: "/map"
      </rosparam>
    </node>

    <node pkg="jsk_pcl_ros" type="roi_clipper" name="roi_attention_clipper"
          output="screen">
      <remap from="~input/image" to="$(arg input_image)"/>
      <remap from="~input/camera_info" to="attention_clipper/output" />
    </node>
  </group> <!-- use mask -->
</launch>
